### ISSUE TITLE
explicit module param incorrectly converted to implicit param

### ISSUE THREAD

**Issue by davdar** [agda/agda#2145]
I tried building my code with branch stable-2.5 agda version 2.5.2 and noticed a change in behavior for explicit module parameters. Consider the following code:

```
-- Agda 2.5.2
module Mod where

record Op (A : Set) : Set where
  field
    op : A → A

module _ (A : Set) {{O : Op A}} where
  f : A → A
  f x = x
  open Op O public

-- f : (A : Set) {{O : Op A}} → A → A 
-- op : {A : Set} {{O : Op A}} → A → A
-- shouldn't `op` be the same type as `f`, where the parameter `A` is
-- explicit?
```

In 2.5.1.1 `f` and `op` have the same type with explicit parameter `A`, whereas in 2.5.2 `op` has `A` as an implicit parameter.

**Comment by WolframKahl** [comment]
I think this is due to a language change documented in the release notes:

```
* Projection parameters [Issue #1954].

  When copying a module, projection parameters will now stay hidden
  arguments, even if the module parameters are visible.
  This matches the situation we had for constructors since long.
  Example:

    module P (A : Set) where
      record R : Set where
        field f : A

    open module Q A = P A

  Parameter A is now hidden in R.f:

    test : ∀{A} → R A → A
    test r = R.f r

  Note that a module parameter that corresponds to the record value
  argument of a projection will not be hidden.

    module M (A : Set) (r : R A) where
      open R A r public

    test' : ∀{A} → R A → A
    test' r = M.f r
```

(Since I want to be able to use released versions 2.5.1\* for the time being, this currently keeps me  from trying master with my own code..).

**Comment by davdar** [comment]
Ah, thanks for the pointer to the docs. I'll mark closed.
